x-pack/plugins/observability_solution/infra/public/alerting/inventory/components/metric.tsx

```diff
@@ -164,9 +172,31 @@ export const MetricExpression = ({
     [customMetric, debouncedOnChangeCustom]
   );
 
-  const availablefieldsOptions = metrics.map((m) => {
-    return { label: m.text, value: m.value };
-  }, []);
+  const metricsToRemove: string[] = metrics
+    .reduce(
+      (metricToRemove, currentMetric) => {
+        if (Object.keys(V2ToLegacyMapping).includes(currentMetric.value)) {
+          metricToRemove.push(V2ToLegacyMapping[currentMetric.value as V2MetricType]);
+        }
+        return metricToRemove;
+      },
+      ['']
+    )
+    .filter(Boolean);
```

Comment:
Could this be a map instead?

```suggestion
  const metricsToRemove: metrics
    .map((currentMetric) => {
      return V2ToLegacyMapping[currentMetric.value as V2MetricType];
    })
    .filter((m): m is string => !!m)
```

Reply:
Changed to `map` ✅